scripts/typescript_checks.py

```diff
@@ -82,12 +835,91 @@ def compile_and_check_typescript(config_path: str) -> None:
     assert process.stdout is not None
     error_messages = list(iter(process.stdout.readline, ''))
 
-    if error_messages:
-        print('Errors found during compilation\n')
-        print('\n'.join(error_messages))
-        sys.exit(1)
+    if config_path == STRICT_TSCONFIG_FILEPATH:
+        # Generate file names from the error messages.
+        errors = [x.strip() for x in error_messages]
+        # Remove the empty lines and error explanation lines.
+        prefixes = ('core', 'extension', 'typings')
+        errors = [x for x in errors if x.startswith(prefixes)]
+        # Remove error explanation lines.
+        errors = [x.split('(', 1)[0] for x in errors]
+        # Remove the duplicate occurrences of the file names.
+        errors = list(dict.fromkeys(errors))
+        files_with_errors = sorted(errors)
+
+        # List of missing files that are neither strictly typed nor present in
+        # NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH.
+        files_not_type_strict = []
+        for filename in files_with_errors:
+            if filename not in NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH:
+                files_not_type_strict.append(filename)
+
+        # Add "typings" folder to get global imports while compiling.
+        files_not_type_strict.append('typings')
+
+        # Update "include" field of tsconfig-strict.json with files that are
+        # neither strict typed nor present in
+        # NOT_FULLY_TYPE_STRICT_TSCONFIG_FILEPATH.
+        # Example: List "files_not_type_strict".
+        file_name = os.path.join(os.getcwd(), 'tsconfig-strict.json')
+        with open(file_name, 'r', encoding='utf-8') as f:
+            tsconfig_strict_json_dict = yaml.safe_load(f)
+            tsconfig_strict_json_dict['include'] = files_not_type_strict
+
+        tsconfig_strict_json_dict = (
+            json.dumps(tsconfig_strict_json_dict, indent=2, sort_keys=True))
+
+        with open(file_name, 'w', encoding='utf-8') as f:
+            f.write(tsconfig_strict_json_dict + '\n')
+
+        # Compile tsconfig-strict.json with updated "include" property.
+        os.environ['PATH'] = '%s/bin:' % common.NODE_PATH + os.environ['PATH']
+        validate_compiled_js_dir()
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
+
+        cmd = ['./node_modules/typescript/bin/tsc', '--project', config_path]
+        process = subprocess.Popen(
+            cmd, stdout=subprocess.PIPE, encoding='utf-8')
+
+        if os.path.exists(COMPILED_JS_DIR):
+            shutil.rmtree(COMPILED_JS_DIR)
```

Comment:
Why do we have to do this twice? I'm not too familiar with how ts compilation works, but my intuition would be that it should be predictable whether running `cmd` produces `COMPILED_JS_DIR`

Reply:
Removed, no need.